### PR TITLE
Hotfix/status codes

### LINKED ISSUE #32: Edit error messages
- [x] De-capitalize first letters of the error message
- [x] Add status code to error classes

### PR DESCRIPTION
**Pull request by Shahar-Y**
Changed status codes according to this link:
[https://github.com/grpc/grpc/blob/master/doc/statuscodes.md](url)
And the messages accordingly.
I kept the original order for easier comparison, but it may look better if arranged in ascending order of status codes.
Should close #32

### CODE REVIEW

File: src/utils/errors/application.error.ts [hunk 1]
```diff
@@ -10,19 +10,19 @@ export class ApplicationError extends Error {
...
-    this.code = code || 500;
+    this.message = message || 'UNKNOWN';
+    this.code = code || 2;
   }
 }
 
 export class ServerError extends ApplicationError {
   constructor(message?: string, code?: number) {
-    super(message || 'internal Server Error', code || 500);
+    super(message || 'UNKNOWN: server side error', code || 2);
```
Comment (omrishtam) [inline src/utils/errors/application.error.ts]: There’s no need to prefix the message with the status code’s meaning. The grpc protocol already knows the meaning of the code and putting it in the message too is meaningless and digresses from the main point of the message, which is to add detail.

Please remove the status prefix from each message, only the status code should change.

File: src/utils/errors/folder.ts [hunk 2]
```diff
@@ -5,24 +5,24 @@ import { ClientError } from './application.error';
 
 export class FolderError extends ClientError {
   constructor(message?: string, code?: number) {
-    super(message || 'bad folder error', code || 400);
+    super(message || 'INVALID_ARGUMENT: bad folder error', code || 3);
```
[the remark this answers is not in the document]
Comment (omrishtam) [reply src/utils/errors/folder.ts]: I personally like `invalid` more💁🏼‍♂️

File: src/utils/errors/file.ts [hunk 3]
```diff
@@ -5,42 +5,42 @@ import { ClientError } from './application.error';
 
 export class FileError extends ClientError {
   constructor(message?: string, code?: number) {
-    super(message || 'bad file error', code || 400);
+    super(message || 'bad file error', code || 3);
```
[the remark this answers is not in the document]
Comment (Shahar-Y) [reply src/utils/errors/file.ts]: mean? evil? horrible? devilish? obnoxious? terrible?

[the remark this answers is not in the document]
Comment (Shahar-Y) [reply src/utils/errors/file.ts]: Actually, I think it is ok to write it like that. 
Anyways, do you prefer the word 'wrong'?

[the remark this answers is not in the document]
Comment (omerzamir) [reply src/utils/errors/file.ts]: `invalid`.

File: src/utils/errors/client.error.ts [hunk 4]
```diff
@@ -2,42 +2,42 @@ import { ClientError } from './application.error';
 
 export class IdInvalidError extends ClientError {
   constructor(message?: string) {
-    super(message || 'id is invalid', 400);
+    super(message || 'id is invalid', 3);
```
Comment (omerzamir) [inline src/utils/errors/client.error.ts]: `invalid id`

File: src/utils/errors/client.error.ts [hunk 5]
```diff
@@ -2,42 +2,42 @@ import { ClientError } from './application.error';
...
   }
 }
 
 export class NameInvalidError extends ClientError {
   constructor(message?: string) {
-    super(message || 'name is invalid', 400);
+    super(message || 'name is invalid', 3);
```
Comment (omerzamir) [inline src/utils/errors/client.error.ts]: `invalid name`

File: src/utils/errors/client.error.ts [hunk 6]
```diff
@@ -2,42 +2,42 @@ import { ClientError } from './application.error';
...
   }
 }
 
 export class MailInvalidError extends ClientError {
   constructor(message?: string) {
-    super(message || 'mail is invalid', 400);
+    super(message || 'mail is invalid', 3);
```
Comment (omerzamir) [inline src/utils/errors/client.error.ts]: `invalid mail`

File: src/utils/errors/client.error.ts [hunk 7]
```diff
@@ -2,42 +2,42 @@ import { ClientError } from './application.error';
...
   constructor(message?: string) {
-    super(message || 'upload not found', 404);
+    super(message || 'upload not found', 5);
   }
 }
 
 export class KeyAlreadyExistsError extends ClientError {
   constructor(key:string, message?: string) {
-    super(message || `the given key ${key} is already in use`, 403);
+    super(message || `the given key ${key} is already in use`, 3);
```
Comment (omerzamir) [inline src/utils/errors/client.error.ts]: `key: ${key} is already in use`

File: src/utils/errors/client.error.ts [hunk 8]
```diff
@@ -2,42 +2,42 @@ import { ClientError } from './application.error';
...
   }
 }
 
 export class FileExistsWithSameName extends ClientError {
   constructor(message?: string) {
-    super(message || 'there is already a file with this name in the folder, try to change the name or the folder', 403);
+    super(message || 'there is already a file with this name in the folder, try to change the name or the folder', 3);
```
Comment (omerzamir) [inline src/utils/errors/client.error.ts]: `file already exist in the folder`

File: src/utils/errors/folder.ts [hunk 9]
```diff
@@ -5,24 +5,24 @@ import { ClientError } from './application.error';
...
   constructor(message?: string, code?: number) {
-    super(message || 'bad folder error', code || 400);
+    super(message || 'invalid folder error', code || 3);
   }
 }
 
 export class FolderNotFoundError extends FolderError {
   constructor(message?: string) {
-    super(message || 'the folder requested was not found', 404);
+    super(message || 'the folder requested was not found', 5);
```
Comment (omerzamir) [inline src/utils/errors/folder.ts]: `requested folder not found`

**Comment by Shahar-Y** [comment]
May I merge?

**Comment by omrishtam** [comment]
> May I merge?

Not yet, we need to change something in api-gw to return the correct status code

**Pull request merged**